evalml/tests/component_tests/test_simple_imputer.py

```diff
@@ -269,38 +270,54 @@ def test_simple_imputer_with_none():
                       "all None": [None, None, None, None]})
     y = pd.Series([0, 0, 1, 0, 1])
     imputer = SimpleImputer()
     imputer.fit(X, y)
     transformed = imputer.transform(X, y)
     expected = pd.DataFrame({"category with None": pd.Series(["b", "a", "a", "a"], dtype='category'),
-                             "boolean with None": pd.Series([True, True, False, True], dtype='boolean'),
+                             "boolean with None": pd.Series([True, True, False, True], dtype='category'),
                              "object with None": pd.Series(["b", "a", "a", "a"], dtype='category')})
-    assert_frame_equal(expected, transformed.to_dataframe(), check_dtype=False)
+    assert_frame_equal(expected, transformed, check_dtype=False)
+
+
+def test_simple_imputer_supports_natural_language():
+    X = pd.DataFrame({"cat with None": ["a", "b", "a", None],
+                      "natural language col": ["free-form text", "will", "be imputed", None]})
+    y = pd.Series([0, 0, 1, 0, 1])
+    X.ww.init(logical_types={"natural language col": "NaturalLanguage"})
+    imputer = SimpleImputer(impute_strategy="constant", fill_value="placeholder")
+    imputer.fit(X, y)
+    transformed = imputer.transform(X, y)
+    expected = pd.DataFrame({"cat with None": pd.Series(["a", "b", "a", "placeholder"], dtype='category'),
+                             "natural language col": pd.Series(["free-form text", "will", "be imputed", "placeholder"], dtype='string')})
+    assert_frame_equal(expected, transformed, check_dtype=False)
 
 
 @pytest.mark.parametrize("X_df", [pd.DataFrame(pd.Series([1, 2, 3], dtype="Int64")),
                                   pd.DataFrame(pd.Series([1., 2., 3.], dtype="float")),
                                   pd.DataFrame(pd.Series(['a', 'b', 'a'], dtype="category")),
-                                  pd.DataFrame(pd.Series([True, False, True], dtype="boolean")),
+                                  pd.DataFrame(pd.Series([True, False, True], dtype=bool)),
                                   pd.DataFrame(pd.Series(['this will be a natural language column because length', 'yay', 'hay'], dtype="string"))])
 @pytest.mark.parametrize("has_nan", [True, False])
-@pytest.mark.parametrize("impute_strategy", ["mean", "median", "most_frequent"])
+@pytest.mark.parametrize("impute_strategy", ["mean", "median"])
 def test_simple_imputer_woodwork_custom_overrides_returned_by_components(X_df, has_nan, impute_strategy):
     y = pd.Series([1, 2, 1])
     if has_nan:
         X_df.iloc[len(X_df) - 1, 0] = np.nan
     override_types = [Integer, Double, Categorical, NaturalLanguage, Boolean]
     for logical_type in override_types:
         try:
-            X = ww.DataTable(X_df, logical_types={0: logical_type})
-        except TypeError:
+            X = X_df.copy()
+            X.ww.init(logical_types={0: logical_type})
+            if _get_invalid_schema_message(X, X.ww.schema):
```

Comment:
The problem here is that before, the conversion from categorical with nan to Boolean would fail but now it doesn't. I've brought that up to the ww team and this is a simple workaround. This is not a change in behavior.